lib/Onyx.js

```diff
@@ -1091,20 +1133,22 @@ function merge(key, changes) {
                 const hasChanged = cache.hasValueChanged(key, modifiedData);
 
                 // This approach prioritizes fast UI changes without waiting for data to be stored in device storage.
-                broadcastUpdate(key, modifiedData, hasChanged, 'merge');
+                const updatePromise = broadcastUpdate(key, modifiedData, hasChanged, 'merge');
 
                 // If the value has not changed, calling Storage.setItem() would be redundant and a waste of performance, so return early instead.
                 if (!hasChanged) {
-                    return Promise.resolve();
+                    return updatePromise;
                 }
 
-                return Storage.mergeItem(key, batchedChanges, modifiedData);
+                return Storage.mergeItem(key, batchedChanges, modifiedData)
+                    .then(() => updatePromise);
             } catch (error) {
                 Logger.logAlert(`An error occurred while applying merge for key: ${key}, Error: ${error}`);
+                return Promise.resolve();
```

Comment:
well, this change actually will have some interesting consequences (I _think_ probably good ones) when a storage error happens.

e.g. if something breaks like not able to save the merge to storage because you are out of space then we will resolve the promise (which would allow anything depending on it to complete to continue as well - which sounds good).

But, if I am not mistaken - I think in this case... we can and should still wait for the `updatePromise` to resolve? 🤔 

Unless that throws an error too. And in that case we should return the resolved promise 😄

Reply:
I don't see the difference to be honest :) Looks like Promise.resolve was also returned in this case just right below catch block.

Reply:
Oh duh, ok I missed that line I guess 😄